datafusion/core/src/dataframe.rs

```diff
@@ -1092,24 +1092,15 @@ mod tests {
     #[tokio::test]
     async fn test_distinct_sort_by() -> Result<()> {
         let t = test_table().await?;
-        let plan = t
+        let err = t
             .select(vec![col("c1")])
             .unwrap()
             .distinct()
             .unwrap()
+            // try to sort on some value not present in input to distinct
             .sort(vec![col("c2").sort(true, true)])
```

Comment:
This is a newly added test in https://github.com/apache/arrow-datafusion/pull/5258 

I think the answer is wrong in this case - in particular you can see there are duplicate values of `c1` produced.